### ISSUE TITLE
Forbid incorrect assign pattern

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#789]
# Rule request

## Thesis

We need to enforce:

```python
a += b
```

instead of:

```python
a = a + b
```

The same applies for all other operators: `-`, `&`, `**`, etc
https://rust-lang.github.io/rust-clippy/master/index.html#assign_op_pattern

## Reasoning

The first version is correct and short. This is a consistency rule.

**Comment by Sam-wyre** [comment]
I would like to work on this

**Comment by sobolevn** [comment]
You are assigned!

**Comment by sobolevn** [comment]
@Sam-wyre hey, how's it going? Do you need any help from my side? Feel free to ask for help or drop this task in case you are willing to.

**Comment by slapshin** [comment]
Good evening! Let's I'll do it.

**Comment by sobolevn** [comment]
Thanks a lot, @slapshin! You are on it.

**Comment by slapshin** [comment]
Hi! Should we handle such situations `a = a + b - c` -> `a += b - c`?

**Comment by sobolevn** [comment]
Yes.

**Comment by slapshin** [comment]
Sorry, another question. For example:

* `a = a + b` can be shortened to `a += b`

* `a = b + a` also can be shortened to `a += b`, because `+` is commutative operation. But need some code analyzing. Should linter do it? It can leads to potentials problems on complex expressions.

What do you think?

**Comment by sobolevn** [comment]
No, we don't need to analyse the code. We should turn both `a = a + b` and `a = b + a` into `a += b`

**Comment by kxepal** [comment]
```
>>> a = [1]
>>> b = [2]
>>> a += b
>>> a
[1, 2]
>>> a = [1]
>>> b = [2]
>>> a = b + a
>>> a
[2, 1]
```
` ¯\_(ツ)_/¯ `

**Comment by sobolevn** [comment]
@kxepal great point! I start to agree to that `a = b + a` is a different case, that should not be covered.

**Comment by sobolevn** [comment]
@slapshin see https://github.com/wemake-services/wemake-python-styleguide/pull/852

**Comment by slapshin** [comment]
@sobolevn ok. Thanks!

**Comment by sobolevn** [comment]
It is quite related and you can take some inspiration from it.

**Comment by slapshin** [comment]
Hi! Simple cases (a = a + b, a = a - b,...) are done. There are difficulties with more complex expressions. For example,
```python

a = a + b + c # can be simply converted to a += b + c

a = a + b - c # can be simply converted to a += b - c

a = a - b + c # can't be converted to a -= b + c (means, a = a - (b + c))

a = a * b + c # can't be converted to a *= b + c

a = a * b * c # can be simply converted to a *= b * c
```


Also, there are cases with brackets. I have no big experience in auto code analyzing, but it looks like we need more deep research here. Maybe you can help me with an idea? Maybe already exists something like this in the system?

**Comment by sobolevn** [comment]
Let's start with simple cases. Later we can adjust this rule for more than one math sign.
